.github/actions/publish-nuget/action.yml

```diff
@@ -37,8 +44,24 @@ runs:
 
         Write-Output "NuGet Source: $source"
         Write-Output "::set-output name=source-url::$source"
+
+        $version = @'
+        ${{ inputs.publish-version }}
+        '@.Trim()
+
+        if ($version -eq 'USE_GITHUB_REF_NAME') { $version = $Env:GITHUB_REF_NAME.Trim().TrimStart("v") }
+        if ($version -eq 'USE_GITHUB_RUN_NUMBER') {
+          $tags = git tag | 
+            ? { $_ -match '^v?\d+(\.\d+){0,3}$' } | 
+            % { [version]($_.TrimStart("v")) } | 
+            Sort-Object -Descending
+          $current = if ($tags.Count -gt 0) { $tags[0] } else { [version]"0.0.0" }
+
+          $next = New-Object "System.Version" $current.Major, $current.Minor, ($current.Build + 1)
```

Comment:
Why is this `$current.Build + 1`, not `$current.Build`?

Reply:
It's based on what's written in the wiki [here](https://lombiq.atlassian.net/wiki/spaces/DEV/pages/786857987/Managing+releases+of+open-source+projects#:~:text=then%20always%20use%20the%20next%20patch%20version). The idea is to create a new pre-release version after the current latest release with a strictly monotonically increasing pre-release suffix. The algorithm is:
- find what's the current latest release using the tags (excluding the ones with pre-release suffix)
  - if none found, use "0.0.0".
- increment the Build version (e.g. 4.6.0 => 4.6.1 or 0.0.0 => 0.0.1)
- add the `-preview-{GITHUB_RUN_NUMBER}` suffix.

Reply:
Ah I see, OK then.